src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -109,9 +115,15 @@ export class AccessibleBufferWidget extends DisposableStore {
 				this._editorWidget.updateOptions({ fontFamily: font.fontFamily, fontSize: font.fontSize, lineHeight: font.charHeight ? font.charHeight * font.lineHeight : 1, letterSpacing: font.letterSpacing });
 			}
 		}));
-		this.add(this._xterm.raw.onWriteParsed(async () => {
+		this.add(this._xterm.raw.onScroll(async () => {
 			if (this._isActive()) {
-				await this._updateEditor(true);
+				await this._updateEditor();
+			}
+		}));
+		this.add(this._xterm.raw.onWriteParsed(async () => {
+			// dynamically update the viewport before there's a scroll event
+			if (!this._xterm.raw.buffer.active.baseY && this._isActive()) {
```

Comment:
Don't we want to update the editor when baseY = 0 as well?

Reply:
yes, we do so by way of `onScroll`

Reply:
`!0` = true